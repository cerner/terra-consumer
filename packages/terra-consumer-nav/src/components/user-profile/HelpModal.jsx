import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import NavHelpContent from '../nav-help/NavHelpContent';
import Modal from '../modal/Modal';
import navElementShape from '../../NavPropShapes';
import styles from './UserProfile.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * An array of navigation elements to be displayed in help menu/popup.
   */
  help: PropTypes.arrayOf(PropTypes.shape(
    navElementShape,
    {
      children: PropTypes.arrayOf(PropTypes.shape(
      navElementShape,
        {
          children: PropTypes.array,
        },
      )),
    },
  )),
  /**
   * Boolean value to render modal.
   */
  isOpen: PropTypes.bool,
  /**
   * Required callback function for use by parent component to update state.
   */
  closeModal: PropTypes.func.isRequired,
};

const defaultProps = {
  help: [],
  isOpen: false,
};

const translations = {
  help: 'Help',
};

const HelpModal = ({
 help,
 isOpen,
 closeModal,
 ...customProps
}) => {
  const modalContent = <NavHelpContent className={cx('nav-item-border')} helpContent={help} />;

  return (
    <Modal
      {...customProps}
      isModalOpen={isOpen}
      title={translations.help}
      content={modalContent}
      closeModal={closeModal}
    />);
};

HelpModal.propTypes = propTypes;
HelpModal.defaultProps = defaultProps;

export default HelpModal;
