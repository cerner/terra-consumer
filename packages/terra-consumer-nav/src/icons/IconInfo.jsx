/* eslint-disable */
import React from 'react';
import IconBase from 'terra-icon/lib/IconBase';

const SvgIcon = (customProps) => {
  const attributes = Object.assign({}, customProps);

  return (
    <IconBase {...attributes}>
      <path d="M24,1C11.317,1,1,11.318,1,24s10.317,23,23,23s23-10.318,23-23S36.683,1,24,1z M24,45C12.421,45,3,35.58,3,24S12.421,3,24,3s21,9.42,21,21S35.579,45,24,45z" />
      <circle cx="24" cy="36" r="2" />
      <path
        d="M24,11c-2.642,0-5.375,1.868-6.5,4.442c-0.222,0.506,0.01,1.096,0.516,1.317c0.508,0.222,1.097-0.01,1.316-0.516
        C20.141,14.395,22.147,13,24,13c2.663,0,5,2.336,5,5s-2.337,5-5,5c-0.553,0-1,0.448-1,1v5c0,0.552,0.447,1,1,1s1-0.448,1-1v-4.075
        c3.341-0.502,6-3.47,6-6.925C31,14.206,27.794,11,24,11z"
      />
    </IconBase>
  );
};

SvgIcon.displayName = "IconInfo";
SvgIcon.defaultProps = {"viewBox":"0 0 48 48","xmlns":"http://www.w3.org/2000/svg","data-name":"Layer 1"};

export default SvgIcon;
/* eslint-enable */
