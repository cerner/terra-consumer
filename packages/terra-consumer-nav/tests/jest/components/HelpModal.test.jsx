import React from 'react';
import HelpModal from '../../../src/components/user-profile/HelpModal';
import IconInfo from '../../../src/icons/IconInfo';

const helpItems = [
  {
    text: 'Technical Questions',
    uri: 'http://localhost:8083/',
    icon: (<IconInfo />),
    children: [{
      text: 'Need help using this portal or need to report an issue? Contact the support team at 123-xxx-xxxx',
      uri: 'http://localhost:8080/',
    }],
  },
  {
    text: 'Get Support ID',
    uri: 'http://localhost:8080/',
    icon: (<IconInfo />),
    children: [],
  },
];

const mockFn = jest.fn();

describe('HelpModal with list of content items', () => {
  // Snapshot Tests
  it('should render a modal with title,close button and content', () => {
    const wrapper = shallow(<HelpModal help={helpItems} closeModal={mockFn} />);
    expect(wrapper).toMatchSnapshot();
  });
});
