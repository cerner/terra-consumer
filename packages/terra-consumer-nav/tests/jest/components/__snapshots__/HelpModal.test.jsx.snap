// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpModal with list of content items should render a modal with title,close button and content 1`] = `
<Modal
  closeModal={[Function]}
  content={
    <NavHelpContent
      className="nav-item-border"
      helpContent={
        Array [
          Object {
            "children": Array [
              Object {
                "text": "Need help using this portal or need to report an issue? Contact the support team at 123-xxx-xxxx",
                "uri": "http://localhost:8083/",
              },
            ],
            "icon": <IconInfo
              data-name="Layer 1"
              viewBox="0 0 48 48"
              xmlns="http://www.w3.org/2000/svg"
        />,
            "text": "Technical Questions",
            "uri": "http://localhost:8083/",
          },
          Object {
            "children": Array [],
            "icon": <IconInfo
              data-name="Layer 1"
              viewBox="0 0 48 48"
              xmlns="http://www.w3.org/2000/svg"
        />,
            "text": "Get Support ID",
            "uri": "http://localhost:8083/",
          },
        ]
      }
    />
  }
  isModalOpen={false}
  title="Help"
/>
`;
